### ISSUE TITLE
[Typescript] [Examples] Mistake with methods bindings

### ISSUE THREAD

**Issue by Shiroy** [photonstorm/phaser#2609]
Hi !

This Issue is about 
- A bug in the API
- A problem with my own code

In this code

``` typescript
class SimpleGame {

    constructor() {
        this.game = new Phaser.Game(800, 600, Phaser.AUTO, 'content', { preload: this.preload, create: this.create });
    }

    game: Phaser.Game;

    preload() {
        this.game.load.image('logo', 'phaser2.png');
    }

    create() {
        var logo = this.game.add.sprite(this.game.world.centerX, this.game.world.centerY, 'logo');
        logo.anchor.setTo(0.5, 0.5);
    }

}

window.onload = () => {

    var game = new SimpleGame();

};
```

If you rename the attribute game to anything else, the code won't pass the preload state because of an undefined attribute 'new game's name'.

This is because, when phaser calls state functions, the `this` attributes is overridden by a `Phaser.Game` object. Because a Phaser.Game has the attribute `game`, it runs perfectly in this case. It crashes in any other cases.

The expected behavior is to have a `this` of type `SimpleGame`.

You can find more information here : https://github.com/Microsoft/TypeScript/wiki/'this'-in-TypeScript

To sum up the article :

``` typescript
var x = this.preload
//is equivalent to
var x = SimpleGame.prototype.preload

// =================
//What is expected
x();
//equivalent to
this.preload();
//==================

//How to get it
var x = SimpleGame.prototype.preload.bind(this);
```

Thanks

**Comment by joelrobichaud** [comment]
The way I see it, this is not a problem with the API, but mostly a problem with the way your code is layed out. Since Typescript uses lexical this, I think the proper way of implementing what you want is by having your classes extend the base Phaser classes, like so

```
class MainState extends Phaser.State {
  preload() {
    // Your state preload code here.
  }

  create() {
    // Your state create code here.
  }
}

class SimpleGame extends Phaser.Game {
  constructor() {
    super({
      width: 800,
      height: 600,
      renderer: Phaser.AUTO,
      parent: 'content'
    });

    this.state.add('main', MainState);
    this.state.start('main');
  }
}

window.onload = () => {
  let game = new SimpleGame();
};
```

**Comment by photonstorm** [comment]
Thank you for taking the time to open this issue. However, official support for Phaser 2 has now ended as we focus on preparing Phaser 3 for beta release. In a bid to keep the project tidy, and avoid confusion, we are closing all v2 related issues.

This does not mean your issue won't be looked-at. In November 2016 the [Phaser CE (Community Edition)](https://github.com/photonstorm/phaser-ce) project began. The aim of Phaser CE was to allow the community to continue v2 development. Since then it has gained significant traction, with many updates and releases.

If you are still interested in having your issue investigated we strongly recommend testing the latest Phaser CE release. And only if the problem persists, opening an issue on the Phaser CE repo.
